clang/test/CodeGen/amdgpu-address-spaces.cpp

```diff
@@ -0,0 +1,31 @@
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --check-globals --version 3
+// RUN: %clang_cc1 -cc1 -triple amdgcn-amd-amdhsa -emit-llvm %s -o - | FileCheck %s
+
+int [[clang::opencl_global]] a = 0;
+int [[clang::opencl_generic]] b = 0;
+int [[clang::opencl_constant]] c = 0;
+[[clang::loader_uninitialized]] int [[clang::opencl_local]] d;
+[[clang::loader_uninitialized]] int [[clang::opencl_private]] e;
+
```

Comment:
maybe test some random numbers too

Reply:
As in, assigning the variables to some random value that's non-zero, or numerical address space attributes.

Reply:
Numbered address spaces

Reply:
Done

Reply:
I didn't mean the initialization value, those are irrelevant. I meant

t [[clang::address_space(999)]] z = 1234;

Reply:
Oh, I figured that it was irrelevant to this change since they already worked, and are probably malformed, but I can add some.